up_client_socket/rust/u_transport_socket.rs

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use async_std::io;
+use async_trait::async_trait;
+use tokio::runtime::Runtime;
+use up_rust::UListener;
+use up_rust::{UAttributesValidators, UriValidator};
+use up_rust::{UCode, UMessage, UMessageType, UStatus, UTransport, UUri};
+
+use protobuf::Message;
+use std::net::TcpStream as TcpStreamSync;
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+use std::io::{Read, Write};
+
+use crate::constants::BYTES_MSG_LENGTH;
+use crate::constants::DISPATCHER_ADDR;
+
+pub trait UtransportExt {
+     fn socket_init(&mut self);
+     fn _handle_publish_message(&mut self, umsg: UMessage);
+     fn _handle_request_message(&mut self, umsg: UMessage);
+    fn read_socket(&self, buffer: &mut [u8]) -> io::Result<usize>;
+}
+
+pub struct UtransportSocket {
+    socket_sync: TcpStreamSync,
+    listner_map: Arc<Mutex<HashMap<String, Vec<Arc<dyn UListener>>>>>,
```

Comment:
Recommend storing instead as:
```suggestion
    listner_map: Arc<Mutex<HashMap<String, HashSet<ComparableListener>>>,
```
so that look up / deleting is cheap and the details of how the hashing / comparisons are done is left behind the veil of [`ComparableListener`](https://github.com/eclipse-uprotocol/up-rust/blob/9efa98059a4170005192104c1eac026b9636e50e/src/utransport.rs#L285).

You can check out the example code [here](https://github.com/eclipse-uprotocol/up-rust/blob/9efa98059a4170005192104c1eac026b9636e50e/src/utransport.rs#L338).